x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockFactory.java

```diff
@@ -481,6 +481,32 @@ public final AggregateMetricDoubleBlock newAggregateMetricDoubleBlock(
         return new AggregateMetricDoubleArrayBlock(min, max, sum, count);
     }
 
+    public DateRangeBlockBuilder newDateRangeBlockBuilder(int estimatedSize) {
+        return new DateRangeBlockBuilder(estimatedSize, this);
+    }
+
+    public DateRangeBlock newConstantDateRangeBlock(DateRangeBlockBuilder.DateRangeLiteral value, int positions) {
+        try (DateRangeBlockBuilder builder = newDateRangeBlockBuilder(positions)) {
+            for (int i = 0; i < positions; i++) {
+                if (value.from() != null) {
+                    builder.from().appendLong(value.from());
+                } else {
+                    builder.appendNull();
+                }
+                if (value.to() != null) {
+                    builder.to().appendLong(value.to());
+                }
+            }
+            return builder.build();
+        }
```

Comment:
I think we'd be better off doing a `DateRangeBlock` that contains to constant blocks.